### PR TITLE
Add Detailed Charge State field

### PR DESCRIPTION
**Pull request by agbpatro**
# Description
Adding a detailed charge state in the payload
If the value is present in string, convert into respective enum

Fixes # (issue)
https://github.com/teslamotors/fleet-telemetry/issues/123

## Type of change

Please select all options that apply to this change:

- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [X] My code follows the style of this project.
- [X] I have performed a self-review of my code.
- [X] I have made corresponding updates to the documentation.
- [X] I have added/updated unit tests to cover my changes.
- [ ] I have added/updated integration tests to cover my changes.

### CODE REVIEW

**Review by aaronpkahn — APPROVED** [review]

**Comment by jbanyer** [comment]
FYI I tried to use this new field, however when I attempt to reference this field in a vehicle configuration (POST /api/1/vehicles/fleet_telemetry_config) I receive the following error:

```
{
  response: null,
  error: 'Unknown field DetailedChargeState',
  error_description: '',
  txid: 'd4cd7d67edf5e099f0f4e46d7e7af102'
}
```

File: telemetry/record.go [hunk 1]
```diff
@@ -225,6 +231,28 @@ func transformLocation(message *protos.Payload) {
...
+			if strVal := datum.GetValue().GetStringValue(); strVal != "" {
+				datum.Value = &protos.Value{Value: &protos.Value_ChargingValue{ChargingValue: parseDetailedChargeState(strVal)}}
+			}
+			// There can be only one Field_DetailedChargeState Datum in the proto; abort once we've seen it.
+			return
+		}
+	}
+}
+
+// parseDetailedChargeState parses a location string (such as "Disconnected") into a *proto.Location type.
```
Comment (bramp) [inline telemetry/record.go:247]: The comment is slightly incorrect, "parses a location", should be "parses a charge state string"....

**Comment by stx** [comment]
@agbpatro Any ETA on this? Super important field.

**Comment by patrickdemers6** [comment]
Closing this as it's addressed in #206. Vehicles will be capable of sending this field in a future firmware release.

**Pull request closed without merging**